### PR TITLE
[Desktop-lite] - VNC_RESOLUTION unreadable fix - issue #945 fix

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Feature name**:
 
 * Desktop-lite
 
 **Description**:
 
 This PR introduces the following functionality :
 
 * This PR is as a solution to Features issue #945 
 * VNC_RESOLUTION's value was not able to be read when set inside `containerEnv` as a configured option, appearing to be available as an Environment variable during build time, but it can only be read at runtime of container.
  
 
 _Changelog_:
 
 * Updated `install.sh`
 * Added a single quotes wrapper around EOF so as to disable variable expansion during build time, and such that the variable's values can be read when they are needed i.e. during runtime of container.
 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

File: src/desktop-lite/install.sh [hunk 1]
```diff
@@ -293,7 +293,7 @@ echo -e "\nSuccess!\n"
 EOF
 
 # Container ENTRYPOINT script
-cat << EOF > /usr/local/share/desktop-init.sh
+cat << 'EOF' > /usr/local/share/desktop-init.sh
```
Comment (samruddhikhandale) [inline src/desktop-lite/install.sh:296]: Can we update patch version?

File: src/desktop-lite/install.sh [hunk 2]
```diff
@@ -293,7 +293,7 @@ echo -e "\nSuccess!\n"
 EOF
 
 # Container ENTRYPOINT script
```
Comment (samruddhikhandale) [inline src/desktop-lite/install.sh:295]: Did you test if it works locally?

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/desktop-lite/install.sh (hunk 1 above)
Comment (gauravsaini04) [reply src/desktop-lite/install.sh:296]: done

File: src/desktop-lite/install.sh (hunk 2 above)
Comment (gauravsaini04) [reply src/desktop-lite/install.sh:295]: yes

**Review by samruddhikhandale — APPROVED** [review]
Thanks! ⚡

**Comment by samruddhikhandale** [comment]
@gauravsaini04 Can you look into test failures?

**Comment by gauravsaini04** [comment]
> @gauravsaini04 Can you look into test failures?

Hi @samruddhikhandale ,
have tested locally with a couple of these scenarios and they seem to be passing the `log-exists` test there in the codespace

File: src/desktop-lite/install.sh (hunk 2 above)
Comment (samruddhikhandale) [inline src/desktop-lite/install.sh:295]: How are the tests passing locally? 

From https://github.com/devcontainers/features/actions/runs/9719274381/job/27190519697?pr=1024#step:4:3497

It looks like these PR changes are failing the init script run. Can you investigate here? 
 
image

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/desktop-lite/install.sh [hunk 3]
```diff
@@ -309,50 +309,50 @@ export LANGUAGE="${LANGUAGE:-"en_US.UTF-8"}"
...
+        echo "$1 is already running." | sudoIf tee -a /tmp/\$1.log > /dev/null
+        log "$1 is already running."
     fi
 }
 
 # Keep command running in background
 keepRunningInBackground()
 {
-    (\$2 bash -c "while :; do echo [\\\$(date)] Process started.; \$3; echo [\\\$(date)] Process exited!; sleep 5; done 2>&1" | sudoIf tee -a /tmp/\$1.log > /dev/null & echo "\$!" | sudoIf tee /tmp/\$1.pid > /dev/null)
+    ($2 bash -c "while :; do echo [\\$(date)] Process started.; $3; echo [\\$(date)] Process exited!; sleep 5; done 2>&1" | sudoIf tee -a /tmp/\$1.log > /dev/null & echo "\$!" | sudoIf tee /tmp/\$1.pid > /dev/null)
```
Comment (samruddhikhandale) [inline src/desktop-lite/install.sh:329]: Good to see that the tests are passing, thanks!  🎉 

Can you now look into why we are seeing this? 👇 
https://github.com/devcontainers/features/actions/runs/10202059742/job/28225358778?pr=1024#step:4:3407
image

Also, can you paste the output of the /tmp/container-init.log file? Also, update the tests to do a cat on the file, would like the see the status there.

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/desktop-lite/install.sh (hunk 3 above)
Comment (gauravsaini04) [reply src/desktop-lite/install.sh:329]:
```
#!/bin/bash

user_name="${USERNAME}"
group_name="$(id -gn ${USERNAME})"
LOG=/tmp/container-init.log

export DBUS_SESSION_BUS_ADDRESS="${DBUS_SESSION_BUS_ADDRESS:-"autolaunch:"}"
export DISPLAY="${DISPLAY:-:1}"
export VNC_RESOLUTION="${VNC_RESOLUTION:-1440x768x16}" 
export LANG="${LANG:-"en_US.UTF-8"}"
export LANGUAGE="${LANGUAGE:-"en_US.UTF-8"}"

# Execute the command it not already running
startInBackgroundIfNotRunning()
{
    log "Starting $1."
    echo -e "\n** $(date) **" | sudoIf tee -a /tmp/\$1.log > /dev/null
    if ! pgrep -x $1 > /dev/null; then
        keepRunningInBackground "$@"
        while ! pgrep -x $1 > /dev/null; do
            sleep 1
        done
        log "$1 started."
    else
        echo "$1 is already running." | sudoIf tee -a /tmp/\$1.log > /dev/null
        log "$1 is already running."
    fi
}

# Keep command running in background
keepRunningInBackground()
{
    ($2 bash -c "while :; do echo [\\$(date)] Process started.; $3; echo [\\$(date)] Process exited!; sleep 5; done 2>&1" | sudoIf tee -a /tmp/\$1.log > /dev/null & echo "\$!" | sudoIf tee /tmp/\$1.pid > /dev/null)
}

# Use sudo to run as root when required
sudoIf()
{
    if [ "$(id -u)" -ne 0 ]; then
        sudo "$@"
    else
        "$@"
    fi
}

# Use sudo to run as non-root user if not already running
sudoUserIf()
{
    if [ "$(id -u)" -eq 0 ] && [ "${user_name}" != "root" ]; then
        sudo -u "${user_name}" "$@"
    else
        "$@"
    fi
}

# Log messages
log()
{
    echo -e "[$(date)] $@" | sudoIf tee -a $LOG > /dev/null
}

log "** SCRIPT START **"

# Start dbus.
log 'Running "/etc/init.d/dbus start".'
if [ -f "/var/run/dbus/pid" ] && ! pgrep -x dbus-daemon  > /dev/null; then
    sudoIf rm -f /var/run/dbus/pid
fi
sudoIf /etc/init.d/dbus start 2>&1 | sudoIf tee -a /tmp/dbus-daemon-system.log > /dev/null
while ! pgrep -x dbus-daemon > /dev/null; do
    sleep 1
done

# Startup tigervnc server and fluxbox
sudoIf rm -rf /tmp/.X11-unix /tmp/.X*-lock
mkdir -p /tmp/.X11-unix
sudoIf chmod 1777 /tmp/.X11-unix
sudoIf chown root:${group_name} /tmp/.X11-unix
if [ "$(echo "${VNC_RESOLUTION}" | tr -cd 'x' | wc -c)" = "1" ]; then VNC_RESOLUTION=${VNC_RESOLUTION}x16; fi
screen_geometry="${VNC_RESOLUTION%*x*}"
screen_depth="${VNC_RESOLUTION##*x}"

# Check if VNC_PASSWORD is set and use the appropriate command
common_options="tigervncserver \${DISPLAY} -geometry ${screen_geometry} -depth ${screen_depth} -rfbport ${VNC_PORT} -dpi ${VNC_DPI:-96} -localhost -desktop fluxbox -fg"

if [ -n "${VNC_PASSWORD+x}" ]; then
    startInBackgroundIfNotRunning "Xtigervnc" sudoUserIf "${common_options} -passwd /usr/local/etc/vscode-dev-containers/vnc-passwd"
else
    startInBackgroundIfNotRunning "Xtigervnc" sudoUserIf "${common_options} -SecurityTypes None"
fi

# Spin up noVNC if installed and not running.
if [ -d "/usr/local/novnc" ] && [ "$(ps -ef | grep /usr/local/novnc/noVNC*/utils/launch.sh | grep -v grep)" = "" ]; then
    keepRunningInBackground "noVNC" sudoIf "/usr/local/novnc/noVNC*/utils/launch.sh --listen ${NOVNC_PORT} --vnc localhost:${VNC_PORT}"
    log "noVNC started."
else
    log "noVNC is already running or not installed."
fi

# Run whatever was passed in
log "Executing \"\$@\"."
exec "$@"
log "** SCRIPT EXIT **"
```

Comment (gauravsaini04) [reply src/desktop-lite/install.sh:329]:
```
root@e41698c17c02:/# cat /tmp/container-init.log
[Fri 02 Aug 2024 08:16:09 AM UTC] ** SCRIPT START **
[Fri 02 Aug 2024 08:16:09 AM UTC] Running "/etc/init.d/dbus start".
[Fri 02 Aug 2024 08:16:09 AM UTC] Starting Xtigervnc.
```

Comment (samruddhikhandale) [reply src/desktop-lite/install.sh:329]: Hmm.. the log seems fine. Thanks! 

@gauravsaini04 Then do you know why we are seeing `/usr/local/share/desktop-init.sh: line 50: sudo: command not found` ? 

Can you check if we see the same message when we run tests on `main`? Also, can you compare the `/tmp/container-init.log` logs as well? 

Want to make sure we don't break things due to this change!

**Review by samruddhikhandale — APPROVED** [review]
Looks great to me!

**Pull request merged**